drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/NegationOperator.java

```diff
@@ -14,6 +16,14 @@ public NegationOperator(ConstraintOperator toBeNegated) {
 
     @Override
     public <T, V> BiPredicate<T, V> asPredicate() {
+        if (toBeNegated instanceof RulebookConstraintOperator) {
+            return (t, v) -> {
+                // if not compatible type, return false regardless of the result
+                // but let the operator be executed anyway so that the error message specific to the constraint is logged
+                boolean result = !toBeNegated.asPredicate().test(t, v);
+                return isCompatibleType(t, v) ? result : false;
```

Comment:
Minor performance improvement: isn't necessary to compute the result if types aren't compatible.

Reply:
I wanted to let the operator log the type check warning. But yes, I can refactor it to make it cleaner.